server/aesdsocket.c

```diff
@@ -0,0 +1,202 @@
+        return -1;
+
+    int yes = 1;
+    if (setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof yes) == -1)
+    {
+        perror("setsockopt");
+        exit(1);
+    }
+
+    if (bind(sockfd, servinfo->ai_addr, servinfo->ai_addrlen) == -1)
+        return -1;
+    // -------------------------------Create-Daemon--after-bind-------------------------------------------------------------------------
+    if (argc > 1)
+    {
+        char *daemon = "-d";
+        if (strcmp(argv[argc - 1], daemon) == 0)
+        {
+            pid_t pid;
+            pid = fork();
+
+            if (pid == -1)
+                return -1;
+            else if (pid != 0)
+                exit(EXIT_SUCCESS);
+
+            if (setsid() == -1)
+                return -1;
+
+            if (chdir("/") == -1)
+                return -1;
+
+            open("/dev/null", O_RDWR);
+            dup(0);
+            dup(0);
+        }
+    }
+
+    // -------------------------------Listen--------------------------------------------------------------------------------------------
+    if (listen(sockfd, 5) == -1)
+        return -1;
+
+    fd = open(file, O_RDWR | O_CREAT | O_TRUNC, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+
+    // In case of an error opening the file
+    if (fd == -1)
+    {
+        syslog(LOG_ERR, "Error opening file with error: %d", errno);
+    }
+
+    while (1)
+    {
+        addr_size = sizeof test_addr;
+        new_conn = accept(sockfd, (struct sockaddr *)&test_addr, &addr_size);
+
+        struct sockaddr_in *p = (struct sockaddr_in *)&test_addr;
+        syslog(LOG_DEBUG, "Accepted connection from %s", inet_ntop(AF_INET, &p->sin_addr, addrstr, sizeof(addrstr)));
+
+        while (1)
+        {
+            bytes_from_client = recv(new_conn, buffer, sizeof(buffer), 0);
+
+            // When no bytes are sent, connection is closed
+            if (bytes_from_client == 0)
+                break;
+
+            // Check if the received packet ends with newline
+            if (buffer[bytes_from_client - 1] == '\n')
+                newlineflag = 1;
+
+            memcpy(buf + (buf_size_count  * starter_buf_size), buffer, bytes_from_client);
+
+            if (newlineflag)
+            {
+                bytes_pending = (buf_size_count * starter_buf_size) + bytes_from_client;
+                break;
+            }
+            else
+            {
+                // When there is no newline increase the buffer by reallocation
+                buf = realloc(buf, (buf_size + starter_buf_size));
+                buf_size += starter_buf_size;
+                buf_size_count += 1;
+            }
+        }
+        if (newlineflag)
+        {
+            // computing total file size
+            fsize += write(fd, buf, bytes_pending);
+
+            // Get back to the beginning of the file
+            lseek(fd, 0, SEEK_SET);
+
+            char *read_buffer = malloc(fsize);
+
+            if (read(fd, read_buffer, fsize) == -1)
```

Comment:
@neohere97 what assumption does this code make about the total size of all packets sent?